dask/dataframe/core.py

```diff
@@ -357,6 +357,43 @@ def _rebuild(self, dsk, *, rename=None):
     def _constructor(self):
         return new_dd_object
 
+    @property
+    def divisions(self):
+        """
+        Tuple of ``npartitions + 1`` values, in ascending order, marking the
+        lower/upper bounds of each partition's index. Divisions allow Dask
+        to know which partition will contain a given value, significantly
+        speeding up operations like `loc`, `merge`, and `groupby` by not
+        having to search the full dataset.
+
+        Example: for ``divisions = (0, 10, 50, 100)``, there are three partitions,
+        where the index in each partition contains values [0, 10), [10, 50),
+        and [50, 100], respectively. Dask therefore knows ``df.loc[45]``
+        will be in the second partition.
+
+        When every item in ``divisions`` is ``None``, the divisions are unknown.
+        Most operations can still be performed, but some will be much slower,
+        and a few may fail.
+
+        It is uncommon to set ``divisions`` directly. Instead, use ``set_index``,
+        which sorts and splits the data as needed.
+        See https://docs.dask.org/en/latest/dataframe-design.html#partitions.
+        """
+        return self._divisions
+
+    @divisions.setter
+    def divisions(self, value):
+        if None in value:
+            if any(v is not None for v in value):
+                raise ValueError("`divisions` must be either all null or all non-null")
```

Comment:
This should also be a warning not an Error for this first PR.

Reply:
What warning should be used here? I considered `PendingDeprecationWarning` but to my knowledge divisions with mixed null / non-null values weren't supported before?

Reply:
I don't have a strong feeling, `FutureWarning` or `PendingDeprecationWarning` is fine with me. There wasn't any check before so people could set whatever divisions they wanted right? Not to say that things would work after that, but they could always set the divisions.